### PR TITLE
API calls for package-update-consumer

### PR DESCRIPTION
**Pull request by KPostOffice**
## Related Issues and Dependencies

…

## This introduces a breaking change

- [ ] Yes
- [x] No

## This should yield a new module release

- [ ] Yes
- [ ] No

## This Pull Request implements

A few api calls for updating and querying the database for package-update-job

### CODE REVIEW

File: thoth/storages/graph/postgres.py [hunk 1]
```diff
@@ -3771,6 +3771,41 @@ def create_python_cve_record(
...
+    def get_repositories_using_package(index_url: str, package_name:str):
+        with self._session_scope() as session, session.begin(subtransactions=True):
+            result = (
+                session.query(PythonPackageVersion)
+                .filter(index_url==index_url)
+                .filter(package_name==package_name)
+                .join(PythonRequirementsLock)
+                .join(ExternalSoftwareEnvironment)
+                .join(AdviserRun)
+                .with_entities(origin)
```
Comment (fridex) [inline thoth/storages/graph/postgres.py]: `distinct`?

For this one and `get_repositories_using_package_version` as well.

File: thoth/storages/graph/postgres.py [hunk 2]
```diff
@@ -3771,6 +3771,41 @@ def create_python_cve_record(
...
+            )
+
+    def get_repositories_using_package_version():
```
Comment (fridex) [inline thoth/storages/graph/postgres.py]: Missing docstring, here and for `get_repositories_using_package`.

File: thoth/storages/graph/postgres.py (hunk 1 above)
Comment (KPostOffice) [reply thoth/storages/graph/postgres.py]: Yeah that's probably best because adviser can be run multiple times under the same origin.

Comment (KPostOffice) [reply thoth/storages/graph/postgres.py]: Hmmmm, this will get old stacks as well...

Comment (KPostOffice) [reply thoth/storages/graph/postgres.py]: I made some changes... I think it should be closer to the intended functionality now.

File: thoth/storages/graph/postgres.py [hunk 3]
```diff
@@ -3771,6 +3771,52 @@ def create_python_cve_record(
...
+        """Update value of is_missing flag for PythonPackageVersion."""
+        with self._session_scope():
+            (session.query(PythonPackageVersion)
+                .update()
+                .where(index_url == index_url)
+                .where(package_name == package_name)
+                .where(package_version == package_version)
+                .values(is_missing=value))
+
+    def get_repositories_using_package(index_url: str, package_name: str) -> List[str]:
```
Comment (pacospace) [inline thoth/storages/graph/postgres.py]: According to naming conventions for queries, this should have `_all` at the end. Moreover, in the docstring I would add `unique` urls of repos... becuase the `distinct` filter is not optional in this case. Same for `get_repositories_using_package_version`.

**Comment by pacospace** [comment]
> LGTM, @pacospace PTAL.
> 
> BTW it would be also good to publish the naming convention here :)

There is a card on that in Next :D

**Comment by harshad16** [comment]
recheck

**Comment by harshad16** [comment]
recheck

File: thoth/storages/graph/postgres.py [hunk 4]
```diff
@@ -3771,6 +3771,70 @@ def create_python_cve_record(
 
                 return False
 
+    def update_missing_flag_package_version(
+        self,
+        value: bool,
+        index_url: str,
+        package_name: str,
+        package_version: str
```
Comment (fridex) [inline thoth/storages/graph/postgres.py]: Positional arguments in this module are in the following order:

1. package name
2. package version
3. index url

Please move the `value` parameter at the end so it reads as "what needs to be updated with what".

File: thoth/storages/graph/postgres.py [hunk 5]
```diff
@@ -3771,6 +3771,70 @@ def create_python_cve_record(
...
+
+    def get_all_repositories_using_package(self, index_url: str, package_name: str) -> List[str]:
+        """Given package info, get unique urls of repos which have been advised with this package."""
+        with self._session_scope() as session:
+            result = (
+                session.query(PythonPackageVersion)
+                .join(PythonPackageIndex)
+                .join(PythonRequirementsLock)
+                .join(AdviserRun, AdviserRun.user_software_stack_id == PythonRequirementsLock.python_software_stack_id)
+                .order_by(AdviserRun.origin)
```
Comment (fridex) [inline thoth/storages/graph/postgres.py]: Is this ordering required? Here and bellow.

Comment (KPostOffice) [reply thoth/storages/graph/postgres.py]: This one might not be, but the other one definitely is.  `.distinct` gets the first result so we want to make sure that each origin has only one result.  I am trying to ensure that the entry which we get corresponds to the most up to date information which we have on said `origin`.

**Pull request merged**